csharp/Managed/Pd.cs

```diff
@@ -0,0 +1,202 @@
+
+		/// <summary>
+		/// Gets the number of output channels.
+		/// </summary>
+		public int Outputs { get; private set; }
+
+		/// <summary>
+		/// Gets the sample rate.
+		/// </summary>
+		public int SampleRate { get; private set; }
+
+		/// <summary>
+		/// Gets the block size.
+		/// </summary>
+		public int BlockSize {
+			get { return PInvoke.blocksize (); }
+		}
+
+		/// <summary>
+		/// Returns [true] when audio computation is enabled, and [false] when audio computation is disabled.
+		/// </summary>
+		public bool IsComputing { get; private set; }
+
+		/// <summary>
+		/// Initializes a new instance of Pd
+		/// </summary>
+		/// <param name="inputChannels">Number of input channels.</param>
+		/// <param name="outputChannels">Number of output channels.</param>
+		/// <param name="sampleRate">Sample rate for project.</param>
+		public Pd (int inputChannels, int outputChannels, int sampleRate) : this (inputChannels, outputChannels, sampleRate, null)
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of Pd
+		/// </summary>
+		/// <param name="inputChannels">Number of input channels.</param>
+		/// <param name="outputChannels">Number of output channels.</param>
+		/// <param name="sampleRate">Sample rate for project.</param>
+		/// <param name="searchPaths">Paths for Pd to search for externals.</param>
+		public Pd (int inputChannels, int outputChannels, int sampleRate, IEnumerable<string> searchPaths)
+		{
+			Inputs = inputChannels;
+			Outputs = outputChannels;
+			SampleRate = sampleRate;
+			//SetupHooks();
+			PInvoke.libpd_init ();
+			foreach (string path in searchPaths ?? Enumerable.Empty<string>()) {
+				PInvoke.add_to_search_path (path);
+			}
+		}
+
+		private void SetupHooks ()
+		{
+			//PrintHook = new LibPDPrintHook(RaisePrintEvent);
+			//PInvoke.set_printhook(PrintHook);
+
+			//BangHook = new LibPDBangHook(RaiseBangEvent);
+			//PInvoke.set_banghook(BangHook);
+
+			//FloatHook = new LibPDFloatHook(RaiseFloatEvent);
+			//PInvoke.set_floathook(FloatHook);
+
+			//SymbolHook = new LibPDSymbolHook(RaiseSymbolEvent);
+			//PInvoke.set_symbolhook(SymbolHook);
+
+			//ListHook = new LibPDListHook(RaiseListEvent);
+			//PInvoke.set_listhook(ListHook);
+
+			//MessageHook = new LibPDMessageHook(RaiseMessageEvent);
+			//PInvoke.set_messagehook(MessageHook);
+		}
+
+		~Pd ()
+		{
+			Dispose (false);
+		}
+
+		public void Dispose ()
+		{
+			Dispose (true);
+			GC.SuppressFinalize (this);
+		}
+
+		private void Dispose (bool disposing)
+		{
+			if (disposing) {
+				Stop ();
+			}
+		}
+
+		/// <summary>
+		/// Starts audio computation.
+		/// </summary>
+		public void Start ()
+		{
+			PInvoke.init_audio (Inputs, Outputs, SampleRate);
+			SendMessage ("pd", "dsp", 1);
+			IsComputing = true;
+		}
+
+		private void SendMessage (string receiver, string message, params object[] args)
+		{
+			SendArgs (args);
+			var finish = PInvoke.finish_message (receiver, message);
+
+			if (finish != 0) {
```

Comment:
I was more thinking of people familiar with C# and its idioms. Maybe I am missing some crucial point here, and different semantics should be applied.